### PR TITLE
Use milliseconds for ForkChoice onTick

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description
Modified `forkChoice.onTick` method to use milliseconds
Added `spec.getMillisPerSlot` config helper method

## Fixed Issue(s)
related to #5297 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java [hunk 1]
```diff
@@ -182,22 +183,32 @@ public void getSlotStartTime_shouldGetCorrectTimePastGenesis() {
 
   @Test
   void onTick_shouldExitImmediatelyWhenCurrentTimeIsBeforeGenesisTime() {
-    final MutableStore store = mock(MutableStore.class);
+    final MutableStore store = spy(MutableStore.class);
```
Comment (ajsutton) [inline ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java]: We can just use a normal mock here.  I'm not a fan of spies if we can possibly avoid them - it can be hard to follow the logic because you often wind up with an odd mix of mocked responses and real ones so never quite know what the test is actually testing or the behaviour it's getting.

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java [hunk 2]
```diff
@@ -182,22 +183,32 @@ public void getSlotStartTime_shouldGetCorrectTimePastGenesis() {
...
+    forkChoiceUtil.onTick(store, UInt64.valueOf(2000000));
 
-    verify(store, never()).setTimeSeconds(any());
+    verify(store, never()).setTimeMillis(any());
   }
 
   @Test
   void onTick_shouldExitImmediatelyWhenCurrentTimeIsBeforeStoreTime() {
-    final MutableStore store = mock(MutableStore.class);
+    final MutableStore store = spy(MutableStore.class);
```
Comment (ajsutton) [inline ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java]: This one only needs a mock as well.

File: ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java [hunk 3]
```diff
@@ -182,22 +183,32 @@ public void getSlotStartTime_shouldGetCorrectTimePastGenesis() {
...
+  }
+
+  @Test
+  void onTick_setsStoreTimeAndStoreTimeMillis() {
+    final MutableStore store = spy(MutableStore.class);
+    when(store.getGenesisTime()).thenReturn(UInt64.valueOf(3000));
+    when(store.getTimeMillis()).thenReturn(UInt64.valueOf(4000000));
+    forkChoiceUtil.onTick(store, UInt64.valueOf(5000678));
+
+    verify(store).setTimeMillis(UInt64.valueOf(5000678));
```
Comment (ajsutton) [inline ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/util/ForkChoiceUtilTest.java]: I think I'd just move this one to using a stub:
```suggestion
    final TestStoreImpl store = new TestStoreFactory(spec).createGenesisStore();
    final UInt64 expectedMillis = store.getTimeMillis().plus(123456);
    forkChoiceUtil.onTick(store, expectedMillis);

    assertThat(store.getTimeMillis()).isEqualTo(expectedMillis);
```

And maybe the test name is just `onTick_setsStoreTimeMillis`?

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceTest.java [hunk 4]
```diff
@@ -210,14 +225,31 @@ void onBlock_shouldReorgWhenProposerWeightingMakesForkBestChain() {
...
-    importBlock(expectedChainHead);
-    assertThat(recentChainData.getStore().getProposerBoostRoot())
-        .contains(expectedChainHead.getRoot());
+    // advance time into slot
+    if (advanceTimeSlotMillis > 0) {
+      StoreTransaction transaction = recentChainData.startStoreTransaction();
+      UInt64 timeIntoSlotMillis =
+          recentChainData.getStore().getTimeMillis().plus(advanceTimeSlotMillis);
+      transaction.setTimeMillis(timeIntoSlotMillis);
+      transaction.commit().join();
```
Comment (ajsutton) [inline ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceTest.java]: I'd use:
```suggestion
      SafeFutureAssert.safeJoin(transaction.commit());
```

otherwise it's possible for `.join()` to block indefinitely if the future doesn't complete which is incredibly hard to debug.

**Review by ajsutton — APPROVED** [review]
LGTM.

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceTest.java (hunk 4 above)
Comment (StefanBratanov) [reply ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceTest.java]: cool, will change and keep in mind for the future
